main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2159,16 +2159,26 @@ object Classpaths {
         classifiers.toVector
       )
     }
+
   def deliverTask(config: TaskKey[PublishConfiguration]): Initialize[Task[File]] =
     Def.task {
       val _ = update.value
       IvyActions.deliver(ivyModule.value, config.value, streams.value.log)
     }
+
   def publishTask(config: TaskKey[PublishConfiguration],
                   deliverKey: TaskKey[_]): Initialize[Task[Unit]] =
-    Def.task {
-      IvyActions.publish(ivyModule.value, config.value, streams.value.log)
+    Def.taskDyn {
```

Comment:
Note that this will make the previous direct dependencies disappear, and introduces dynamic dependencies that are not visible to `inspect`. Every time I see this I become more convinced that we need a dynamic inspect, as LAMP proposed some time ago.

Reply:
Wow, I was just trying to make sense of the task graph for `publish` with `inspect`, now I know why I can't…